kairon/shared/test/processor.py

```diff
@@ -163,36 +168,71 @@ def get_by_id_and_type(reference_id: str, bot: str, log_type: str, start_idx: in
         @return: list of logs.
         """
         logs = []
-        filtered_data = ModelTestingLogs.objects(reference_id=reference_id, bot=bot, type=log_type)
-        if log_type == 'stories' and filtered_data:
+        filter_log_type = 'stories' if log_type == 'stories' else 'nlu'
+        filtered_data = ModelTestingLogs.objects(reference_id=reference_id, bot=bot, type=filter_log_type)
+        if log_type == LOG_TYPE.stories.value and filtered_data:
             filtered_data = filtered_data.get()
             logs = filtered_data.data.get('failed_stories', [])[start_idx:start_idx+page_size]
             fail_cnt = filtered_data.data.get('conversation_accuracy', {}).get('failure_count', 0)
-            logs = {'errors': logs, 'total': fail_cnt}
+            success_cnt = filtered_data.data.get('conversation_accuracy', {}).get('success_count', 0)
+            total_cnt = filtered_data.data.get('conversation_accuracy', {}).get('total_count', 0)
+            logs = {'errors': logs, 'failure_count': fail_cnt, 'success_count': success_cnt, 'total_count': total_cnt}
             if fail_cnt:
                 logs = json.dumps(logs)
                 logs = json.loads(logs)
-        elif log_type == 'nlu' and filtered_data:
-            intent_failures, entity_failures, response_selection_failures = [], [], []
-            intent_failure_cnt, entity_failure_cnt, response_selection_failure_cnt = 0, 0, 0
+        elif log_type == LOG_TYPE.nlu.value and filtered_data:
+            intent_failures = []
+            intent_failure_cnt, intent_success_cnt, intent_total_cnt = 0, 0, 0
             filtered_data = filtered_data.get()
             if filtered_data.data.get('intent_evaluation') and filtered_data.data['intent_evaluation'].get('errors'):
```

Comment:
this will only get failures, success and total count when there are errors. we need to get the counts irrespective of errors.

Reply:
Changes made:
1. retrieved num_failures/num_success/total_tests count from logs
 2. added flag is_augmented into the logs and returned same during logs retrieval
 3. added separate log type for retrieval of entity evaluation and response selection evaluation results
 4. added/modified unit and integration tests for the same.